ref-engine-discovery.md

```diff
@@ -0,0 +1,134 @@
+# OCI Ref-engine Discovery
+
+This is version 0.1 of this specification.
+
+To faciliate communication between image publishers and consumers, this specification defines a [ref-engine discovery](#ref-engine-discovery) protocol which publishers MAY use to direct consumers towards [reference engines](#ref-engine).
+Publishers who choose not to support this specification can safely ignore the remainder of this document.
+
+Having retrieved a set of reference engines (via this and other protocols), consumers can use those ref engines to recover a set of [Merkle roots](#merkle-root) potentially associated with a given image name.
+Consumers who choose not to support this specification can safely ignore the remainder of this document.
+Consumers who choose to support this specification MAY attempt to discover and use ref engines via other channels, and only fall back to this protocol if those ref engines do not return a satisfactory Merkle root.
+
+## Glossary
+
+### Ref-engine discovery
+
+A service that suggests possible [ref engines](#ref-engine).
+This specification defines a ref-engine discovery protocol.
+
+### CAS engine
+
+A service that provides access to [content-addressable storage][cas].
+
+### Merkle root
+
+The root node in a [Merkle tree][Merkle-tree].
+In the OCI ecosystem, Merkle links are made via [descriptors][descriptor].
+The Merkle root may be a descriptor ([media type][media-type] [`application/vnd.oci.descriptor.v1+json`][descriptor]), or it may have a different media type.
+Merkle roots may suggest [CAS engines](#cas-engine), e.g. via a `casEngines` entry in their JSON, but that is out of scope for ref-engine discovery.
+
+### Ref engine
+
+A service that maps an image name to a set of potential [Merkle roots](#merkle-root).
+
+## `oci-host-ref-engines` well-known URI registration
+
+This specification registers the `oci-host-ref-engines` well-known URI in the Well-Known URI Registery as defined by [RFC 5785][rfc5785].
+
+URI suffix: `oci-host-ref-engines`
+
+Change controller: The [Open Container Initiative][OCI]
+
+Specification document(s): This specification
+
+Related information: None
+
+## Images associated with a host's `oci-host-ref-engines`
+
+Publishers SHOULD populate the `oci-host-ref-engines` resource with ref engines which are capable of resolving image names that match the [`host-based-image-name` rule](host-based-image-names.md) with a `host` part that matching their [fully qualified domain name][rfc1594-s5.2] and its subdomains or deeper descendants.
+For example, https://b.example.com/.well-known/oci-host-ref-engines SHOULD prefer ref engines capable of resolving image names with `host` parts matching `b.example.com`, `a.b.example.com`, etc.
```

Comment:
nit: https://b.example.com/.well-known/oci-host-ref-engines should add \' \', otherwise it will be super link unable to visit.